runtime/j9vm/jvm.c

```diff
@@ -4474,12 +4504,28 @@ JVM_RaiseSignal(jint sigNum)
 {
 	jboolean rc = JNI_FALSE;
 	J9JavaVM *javaVM = (J9JavaVM *)BFUjavaVM;
+	BOOLEAN isShutdownSignal = isSignalUsedForShutdown(sigNum);
 
 	Trc_SC_RaiseSignal_Entry(sigNum);
 
-	if (J9_ARE_ALL_BITS_SET(javaVM->sigFlags, J9_SIG_XRS)
-			&& isSignalSpecial(sigNum)) {
-		/* Ignore signal */
+	if (isSignalUsedByVM(sigNum)) {
+		/* JVM_RaiseSignal shouldn't raise synchronous signals such as SIGFPE,
+		 * SIGILL and SIGSEGV. Based upon the spec, it should also not raise SIGQUIT.
+		 */
+	} else if (J9_ARE_ALL_BITS_SET(javaVM->sigFlags, J9_SIG_XRS_ASYNC) && isShutdownSignal) {
+		/* Ignore all shutdown signals if -Xrs or -Xrs:async is specified. */
```

Comment:
None of the code is checking J9_SIG_XRS any more, which doesn't match the comments.

Reply:
Hmm, just noticed that J9_SIG_XRS and J9_SIG_XRS_SYNC have the same value.

Reply:
I wonder if https://www.ibm.com/support/knowledgecenter/en/SSYKE2_8.0.0/com.ibm.java.lnx.80.doc/diag/appendixes/cmdline/Xrs.html meant to document -Xrs:async and not -Xrs:sync.

Reply:
Note there is an explicit J9_SIG_NO_SIG_QUIT as well.
```
		if (optionValue && 0 == strcmp(optionValue, "sync")) {
			vm->sigFlags |= J9_SIG_XRS_SYNC;
			sigOptions |= J9PORT_SIG_OPTIONS_REDUCED_SIGNALS_SYNCHRONOUS;
		} else if (optionValue && 0 == strcmp(optionValue, "async")) {
			vm->sigFlags |= (J9_SIG_XRS_ASYNC | J9_SIG_NO_SIG_QUIT);
			sigOptions |= J9PORT_SIG_OPTIONS_REDUCED_SIGNALS_ASYNCHRONOUS;
		} else {
			vm->sigFlags |= (J9_SIG_XRS_SYNC | J9_SIG_XRS_ASYNC | J9_SIG_NO_SIG_QUIT);
			sigOptions |= (J9PORT_SIG_OPTIONS_REDUCED_SIGNALS_SYNCHRONOUS | J9PORT_SIG_OPTIONS_REDUCED_SIGNALS_ASYNCHRONOUS);
		}
```

Reply:
As an aside, I think we should change the one piece of JIT code that uses J9_SIG_XRS and then delete this duplicate flag.

Reply:
according to `SunMiscSignalTest`, `JVM_RegisterSignal` and `JVM_RaiseSignal` should never register or raise `SIGQUIT`. JVM should manage `SIGQUIT` separately via `J9SigQuitStartup` and `J9SigQuitShutdown`.

I will open a pull request for removal of the duplicate flag.